_universum/modules/launcher.py

```diff
@@ -85,6 +85,75 @@ def check_if_env_set(configuration):
     return True
 
 
+def check_str_match(string, expected_substrings, unexpected_substrings):
+    """The function to check whenever specified string contains 'expected' and
+    does NOT contains 'unexpected' substrings.
+
+    ["", ["parent 1", "parent 2", "parent 1 step 1", "parent 2 step 1", "parent 1 step 2", "parent 2 step 2"], []],
+    >>> check_str_match("step 1", [], [])
+    True
+
+    >>> check_str_match("step 1", ["step 1"], [])
+    True
+
+    >>> check_str_match("step 1", ["step 1"], ["step 1"])
+    False
+
+    >>> check_str_match("step 1", [], ["step 1"])
+    False
+
+    >>> check_str_match("step 1", ["step "], ["1"])
+    False
+
+    :rtype: bool
+    """
+    result = False if expected_substrings else True
+    for substr in expected_substrings:
+        if substr in string:
+            result = True
+            break
+
+    for substr in unexpected_substrings:
+        if substr in string:
+            result = False
+            break
+    return result
+
+
+def get_match_patterns(filters):
+    """The function to parse 'filters' defined as single string into a lists of
+    'expected' and 'unexpected' patterns.
+
+    >>> get_match_patterns("")
+    [], []
+
+    >>> get_match_patterns(":")
+    [], []
+
+    >>> get_match_patterns("!:")
+    [], []
+
+    >>> get_match_patterns("f:")
+    ["f"], []
+
+    >>> get_match_patterns("f:!f 1")
+    ["f"], ["f 1"]
+
+    >>> get_match_patterns("f:!f 1:f 2:!f 3")
+    ["f", "f 2"], ["f 1", "f 3"]
+    """
+    expected = []
+    unexpected = []
+
+    filters = filters.split(':') if filters else []
+    for f in filters:
+        if f.startswith('!'):
+            unexpected.append(f[1:]) if len(f) > 1 else None # pylint: disable=expression-not-assigned
```

Comment:
Do we _really_ need a one-line if in here? Pylint raises a warning here for a reason, a think.

Reply:
we don't want 3-level of nesting